### PR TITLE
Transpile to ES5_STRICT, use un-minified code.

### PR DESCRIPTION
**Pull request by klausw**
This is initially intended for discussion - the change currently uses un-minified Angular unconditionally,
not sure if that's acceptable for performance.

I've also tried to use the minified version with a source map, this now works for source code locations but not for the renamed variables, making debugging nearly impossible. So I wouldn't recommend that variant.

### CODE REVIEW

**Comment by jmuharsky** [comment]
I'm okay with non-performant code for debugging/testing purposes, but for production release we'll want minified and optimized JS.

**Comment by klausw** [comment]
PTAL, I've changed it so that it uses the minified code by default. Overrideable via:

    appcfg update -E PKE_DEBUG:1 deploy

Even in the minified version, it now includes the source files mentioned in the source map, so source is visible.

**Comment by klausw** [comment]
FYI, the local compilation continues to use SIMPLE_OPTIMIZATIONS, I've added PRETTY_PRINT which shouldn't affect performance beyond increased file size. Not sure how to push the conditional into the gulpfile.

File: gulpfile.js [hunk 1]
```diff
@@ -29,23 +29,23 @@ gulp.task('third_party', function() {
       .pipe(gulp.dest('deploy/client/third_party/jquery'));
 
   /** Angular */
-  gulp.src('bower_components/angular/angular.min.*')
+  gulp.src('bower_components/angular/angular.*')
```
Comment (jmuharsky) [inline gulpfile.js]: Gulp has a recipe for dealing with minified vs. non-minified files; this should work well in this case.

https://github.com/gulpjs/gulp/commit/61b5751f46749d3ec4611bb3bf8d07de4faf4f65

File: server/perfkit/explorer/handlers/base.py [hunk 2]
```diff
@@ -90,6 +90,10 @@ def RenderHtml(self, template_file, template_values, status=200):
     template_values['static_dir'] = ('/_static/%s' %
                                      os.environ['CURRENT_VERSION_ID'])
     template_values['env'] = self.env
+    if 'PKE_DEBUG' in os.environ:
```
Comment (jmuharsky) [inline server/perfkit/explorer/handlers/base.py]: Recommend allowing this parameter to be used in the URL as well, so that it can be used on deployments.

**Comment by jmuharsky** [comment]
Is the conditional in this case just to decide which third_party.html file(s) to include?  I don't think it's necessarily the best choice to build this into the Gulp task -- it would be ideal to switch this on the fly from a deployed site.

File: server/perfkit/explorer/handlers/base.py (hunk 2 above)
Comment (klausw) [inline server/perfkit/explorer/handlers/base.py]: done, add &debug=1 to the URL to activate it, or &debug=0 if you want to disable it while PKE_DEBUG is on.

**Comment by klausw** [comment]
No, I had meant to make the gulpfile conditional for the compilation
settings. But this currently only affects the pretty_print setting which
should be ok to leave on unconditionally - we can revisit this once we get
uncompiled client code working. So I think we don't need a gulpfile
conditional at this point.

**Comment by klausw** [comment]
PTAL, The tests now also run strict mode, this discovered a few more pre-existing issues.

**Comment by klausw** [comment]
Rebased to pick up the widgetEditorController test fix, all tests are passing now.

File: client/components/dashboard/dashboard-service.js [hunk 3]
```diff
@@ -794,7 +794,12 @@ DashboardService.prototype.moveWidgetToContainer = function(
   targetContainer.model.container.columns += widget.model.layout.columnspan;
   targetContainer.model.container.children.push(widget);
   widget.state().parent = targetContainer;
-  this.selectedContainer = targetContainer;
+
+  // Move the container selection to the target container. Don't change
```
Comment (jmuharsky) [inline client/components/dashboard/dashboard-service.js]: Recommend calling dashboardService.selectContainer for this.

**Comment by jmuharsky** [comment]
Per discussion, please add documentation on how to set Debug mode on a test, or during deployment.

**Comment by jmuharsky** [comment]
Otherwise LGTM.

**Pull request merged**